src/main/java/com/hrznstudio/galacticraft/block/special/fluidpipe/FluidPipeBlockEntity.java

```diff
@@ -59,25 +58,42 @@ public FluidPipeBlockEntity(BlockEntityType<? extends FluidPipeBlockEntity> enti
     }
 
     @Override
-    public void setLocation(World world, BlockPos pos) {
-        super.setLocation(world, pos);
+    public @NotNull Pipe.FluidData getFluid() {
+        return data;
     }
 
     @Override
-    public @NotNull Pipe.FluidData getData() {
-        return data;
+    public @NotNull PipeNetwork getNetwork() {
+        if (!world.isClient()) {
+            if (this.network == null) {
+                for (Direction direction : Direction.values()) {
+                    BlockEntity entity = world.getBlockEntity(pos.offset(direction));
+                    if (entity instanceof Pipe) {
+                        if (((Pipe) entity).getNetworkNullable() != null) {
+                            ((Pipe) entity).getNetworkNullable().addPipe(pos, this);
+                            break;
+                        }
+                    }
+                }
+                if (this.network == null) {
```

Comment:
wont this always be true because of L68?

Reply:
and shouldn't this stuff be before adding pipes to the network?

Reply:
> wont this always be true because of L68?

L69-77 (or more specifically L73) can set the network, if successful.

> and shouldn't this stuff be before adding pipes to the network?

If you mean creating a network for itself [L79] before adding itself to another network [L73] no, as there's no need to create a network only to never use it (get replaced by a preexisting network)

Reply:
ah cool, thanks for clearing that up for me.